// lottie-react-native/lib/module/LottieView.js
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
import React from 'react';
import { View, StyleSheet, Animated, processColor } from 'react-native';
import NativeLottieAnimationView, { Commands } from './specs/LottieAnimationViewNativeComponent';
const AnimatedNativeLottieView = Animated.createAnimatedComponent(NativeLottieAnimationView);
const defaultProps = {
  source: undefined,
  progress: 0,
  speed: 1,
  loop: true,
  autoPlay: false,
  autoSize: false,
  enableMergePathsAndroidForKitKatAndAbove: false,
  cacheComposition: true,
  useNativeLooping: false,
  resizeMode: 'contain',
  colorFilters: [],
  textFiltersAndroid: [],
  textFiltersIOS: []
};

/**
 * View hosting the lottie animation.
 */
export class AnimatedLottieView extends React.PureComponent {
  constructor(props) {
    super(props);
    _defineProperty(this, "_lottieAnimationViewRef", void 0);
    _defineProperty(this, "onAnimationFinish", evt => {
      if (this.props.onAnimationFinish) {
        this.props.onAnimationFinish(evt.nativeEvent.isCancelled);
      }
    });
    this.play = this.play.bind(this);
    this.reset = this.reset.bind(this);
    this.pause = this.pause.bind(this);
    this.resume = this.resume.bind(this);
    this.onAnimationFinish = this.onAnimationFinish.bind(this);
    this._captureRef = this._captureRef.bind(this);
  }
  play(startFrame, endFrame) {
    Commands.play(this._lottieAnimationViewRef, startFrame ?? -1, endFrame ?? -1);
  }
  reset() {
    Commands.reset(this._lottieAnimationViewRef);
  }
  pause() {
    Commands.pause(this._lottieAnimationViewRef);
  }
  resume() {
    Commands.resume(this._lottieAnimationViewRef);
  }
  _captureRef(ref) {
    if (ref === null) {
      return;
    }
    this._lottieAnimationViewRef = ref;
    if (this.props.autoPlay === true) {
      this.play();
    }
  }
  render() {
    var _this$props$colorFilt;
    const {
      style,
      source,
      autoSize,
      autoPlay,
      duration,
      textFiltersAndroid,
      textFiltersIOS,
      ...rest
    } = this.props;
    const sourceName = typeof source === 'string' ? source : undefined;
    const sourceJson = typeof source === 'object' && !source.uri ? JSON.stringify(source) : undefined;
    const sourceURL = typeof source === 'object' && source.uri ? source.uri : undefined;
    const aspectRatioStyle = sourceJson ? {
      aspectRatio: source.w / source.h
    } : undefined;
    const styleObject = StyleSheet.flatten(style);
    let sizeStyle;
    if (!styleObject || styleObject.width === undefined && styleObject.height === undefined) {
      sizeStyle = autoSize && sourceJson ? {
        width: source.w
      } : StyleSheet.absoluteFill;
    }
    const speed = duration && sourceJson && source.fr ? Math.round(source.op / source.fr * 1000 / duration) : this.props.speed;
    const colorFilters = (_this$props$colorFilt = this.props.colorFilters) === null || _this$props$colorFilt === void 0 ? void 0 : _this$props$colorFilt.map(colorFilter => ({
      ...colorFilter,
      color: processColor(colorFilter.color)
    }));
    return /*#__PURE__*/React.createElement(View, {
      style: [aspectRatioStyle, sizeStyle, style]
    }, /*#__PURE__*/React.createElement(AnimatedNativeLottieView, _extends({
      ref: this._captureRef
    }, rest, {
      colorFilters: colorFilters,
      textFiltersAndroid: textFiltersAndroid,
      textFiltersIOS: textFiltersIOS,
      speed: speed,
      style: [aspectRatioStyle, sizeStyle || {
        width: '100%',
        height: '100%'
      }, style],
      sourceName: sourceName,
      sourceJson: sourceJson,
      sourceURL: sourceURL,
      onAnimationFinish: this.onAnimationFinish,
      autoPlay: autoPlay
    })));
  }
}
_defineProperty(AnimatedLottieView, "defaultProps", defaultProps);
